knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/service/ServiceEngine.java

```diff
@@ -53,12 +56,18 @@ public ServiceEngine(Vertx vertx, TemplateEngineConfiguration serviceConfigurati
                     resp.subscribe(subscriber);
                     req.end();
                 });
-        return serviceResponse.flatMap(response ->
-                Observable.just(Buffer.buffer())
-                        .mergeWith(response.toObservable())
-                        .reduce(Buffer::appendBuffer))
-                .doOnNext(this::traceServiceCall)
-                .flatMap(buffer -> Observable.just(buffer.toJsonObject().getMap()));
+        return serviceResponse.flatMap(response -> Observable.just(Buffer.buffer())
+                .mergeWith(response.toObservable())
+                .reduce(Buffer::appendBuffer)
+                .flatMap(buffer -> {
+                    JsonObject jsonObject = buffer.toJsonObject();
+                    Map<String, String> responseMap = Maps.newHashMap();
+                    responseMap.put("statusCode", String.valueOf(response.statusCode()));
+                    jsonObject.getMap().put("_response", responseMap);
```

Comment:
Don't understand why Map is doing anything here while just getting key=value pair.
- Whole invocation of this flatMap() should be:
```
.flatMap(buffer -> 
      buffer.toJsonObject().put("_response", 
                                               new JsonObject().put("statusCode", response.statusCode()) //you can extract to method called e.g. processHeaders(Buffer) if you want.
).doOnNext(this::traceServiceCall);
```